app/controllers/searches_controller.rb

```diff
@@ -3,7 +3,17 @@ class SearchesController < ApplicationController
 
   def show
     return unless params[:query] && params[:query].is_a?(String)
-    @gems = Rubygem.search(params[:query]).with_versions.paginate(page: @page)
+    if params[:es] == 'true'
+      begin
+        @gems = Rubygem.search(params[:query]).page(@page).records
+        @gems.size
```

Comment:
I think it will be better to move this code block into some separate "concern" (not controllers/concerns) and just call it here in controller to keep controller simple. For example make `Rubygem.search` a "search dispatcher" and send `params[:es] == 'true'` as an argument there. Keep `legacy_search` and add `elastic_search` method and move this responsibility out of controller.